source/agora/consensus/Genesis.d

```diff
@@ -50,14 +50,9 @@ public ref immutable(Block) GenesisBlock () nothrow @safe @nogc
 
 pragma(inline, true)
 public void setGenesisBlock (immutable Block* block)
-    nothrow @trusted @nogc
+    @safe
 {
     assert(block !is null);
```

Comment:
If you were unblocked, it means you can put the validation back here again.

When I suggested to move this into `FullNode`, it was a suggestion in case we didn't have time for #947.

Reply:
It causes the same problem again.
It is being used in an integrated test.

Reply:
That's why I asked you if your PR fixes the issue..

Reply:
I did a test with the existing code.
I have added a new commit to remove unnecessary import.